sklearn/gaussian_process/gpr.py

```diff
@@ -209,12 +220,16 @@ def obj_func(theta, eval_gradient=True):
                         "Multiple optimizer restarts (n_restarts_optimizer>0) "
                         "requires that all bounds are finite.")
                 bounds = self.kernel_.bounds
-                for iteration in range(self.n_restarts_optimizer):
+                theta_initials = []
+                for i in range(self.n_restarts_optimizer):
```

Comment:
use the size parameter of `rng.uniform` to generate n_restarts_optimizer elements in one fast operation